### PR TITLE
Refactor: shim acc test for machinelearning

### PR DESCRIPTION
**Pull request by ArcturusZhang**

### CODE REVIEW

File: azurerm/internal/services/machinelearning/machine_learning_workspace_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+func TestAccAzureRMMachineLearningWorkspace_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_machine_learning_workspace", "test")
+	r := WorkspaceResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
```
Comment (tombuildsstuff) [inline azurerm/internal/services/machinelearning/machine_learning_workspace_resource_test.go:79]: this also needs to check the Identity block too:

```suggestion
				check.That(data.ResourceName).ExistsInAzure(r),
				check.That(data.ResourceName).Key("identity.#").HasValue("1"),
 				check.That(data.ResourceName).Key("identity.0.type").HasValue("SystemAssigned"),
 				check.That(data.ResourceName).Key("identity.0.principal_id").Exists(),
 				check.That(data.ResourceName).Key("identity.0.tenant_id").Exists(),
```

File: azurerm/internal/services/machinelearning/machine_learning_workspace_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.completeUpdated(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
```
Comment (tombuildsstuff) [inline azurerm/internal/services/machinelearning/machine_learning_workspace_resource_test.go:101]: this also needs to check the Identity block too:

```suggestion
				check.That(data.ResourceName).ExistsInAzure(r),
				check.That(data.ResourceName).Key("identity.#").HasValue("1"),
 				check.That(data.ResourceName).Key("identity.0.type").HasValue("SystemAssigned"),
 				check.That(data.ResourceName).Key("identity.0.principal_id").Exists(),
 				check.That(data.ResourceName).Key("identity.0.tenant_id").Exists(),
```

File: azurerm/internal/services/machinelearning/machine_learning_workspace_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+func TestAccAzureRMMachineLearningWorkspace_completeUpdate(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_machine_learning_workspace", "test")
+	r := WorkspaceResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
```
Comment (tombuildsstuff) [inline azurerm/internal/services/machinelearning/machine_learning_workspace_resource_test.go:94]: this also needs to check the Identity block too:

```suggestion
				check.That(data.ResourceName).ExistsInAzure(r),
				check.That(data.ResourceName).Key("identity.#").HasValue("1"),
				check.That(data.ResourceName).Key("identity.0.type").HasValue("SystemAssigned"),
				check.That(data.ResourceName).Key("identity.0.principal_id").Exists(),
				check.That(data.ResourceName).Key("identity.0.tenant_id").Exists(),
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @ArcturusZhang 

Thanks for this PR.

Taking a look through this mostly LGTM, if we can fix up the comments then this should otherwise be good to merge.

Thanks!

File: azurerm/internal/services/machinelearning/machine_learning_workspace_resource_test.go [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+  name                    = "acctest-MLW-%d"
+  location                = azurerm_resource_group.test.location
+  resource_group_name     = azurerm_resource_group.test.name
+  application_insights_id = azurerm_application_insights.test.id
+  key_vault_id            = azurerm_key_vault.test.id
+  storage_account_id      = azurerm_storage_account.test.id
+
+  identity {
+    type = "SystemAssigned"
+  }
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply azurerm/internal/services/machinelearning/machine_learning_workspace_resource_test.go:141]: Hmmm in this resource the identity is Required: https://github.com/terraform-providers/terraform-provider-azurerm/blob/2fb9b056b9a9d9c66ff0d12c59742e3501447890/azurerm/internal/services/machinelearning/machine_learning_workspace_resource.go#L88

File: azurerm/internal/services/machinelearning/machine_learning_workspace_data_source_test.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+				check.That(data.ResourceName).Key("identity.0.type").HasValue("SystemAssigned"),
+				check.That(data.ResourceName).Key("identity.0.principal_id").Exists(),
+				check.That(data.ResourceName).Key("identity.0.tenant_id").Exists(),
+			),
+		},
+	})
+}
+
+func (d WorkspaceDataSource) basic(data acceptance.TestData) string {
+	config := WorkspaceResource{}.basic(data)
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply azurerm/internal/services/machinelearning/machine_learning_workspace_data_source_test.go:32]: Done

File: azurerm/internal/services/machinelearning/machine_learning_workspace_resource_test.go (hunk 1 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/machinelearning/machine_learning_workspace_resource_test.go:79]: Done

**Comment by ArcturusZhang** [comment]
Hi @tombuildsstuff I have resolved the comments, please have a look, thanks

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Pull request merged**